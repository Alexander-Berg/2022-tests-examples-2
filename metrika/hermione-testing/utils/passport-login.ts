declare let hermione: any;

export async function passportLogin() {
    const timeoutMS = 10000;
    const loginInputSelector = '#passp-field-login';
    const passwodInputSelector = '#passp-field-passwd';
    const signInButtonSelector = '.passp-sign-in-button';
    const profileSelector = '.profile';

    const { username, password } = hermione.ctx.user;

    await this.browser.url('https://passport.yandex.ru/');

    // ввести логин и нажать кнопку
    await this.browser.$(loginInputSelector).setValue(username);
    await this.browser.$(signInButtonSelector).click();

    // дождаться загрузки поля для ввода пароля
    await this.browser.waitForExist(passwodInputSelector);

    // ввести пароль и нажать кнопку
    await this.browser.$(passwodInputSelector).setValue(password);
    await this.browser.$(signInButtonSelector).click();

    // дождаться завершения авторизации и перехода на страницу с профилем
    await this.browser.waitForExist(profileSelector, timeoutMS);
}
