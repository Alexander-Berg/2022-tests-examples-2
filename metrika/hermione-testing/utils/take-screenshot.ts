export async function takeScreenshot(
    url: string,
    selector: string,
    screenshotFileName: string,
) {
    await this.browser.url(url);
    await this.browser.waitForVisible(selector);
    await this.browser.assertView(screenshotFileName, selector);
}
