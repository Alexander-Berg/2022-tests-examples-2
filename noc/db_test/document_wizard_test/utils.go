package trigger

func Range(n int) []int {
	result := make([]int, 0)
	for i := 0; i < n; i++ {
		i := i
		result = append(result, i)
	}
	return result
}
