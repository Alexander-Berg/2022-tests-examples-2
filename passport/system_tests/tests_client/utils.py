import random
import string


class Account(object):

    def __init__(self, uid, login, password, firstname=None, lastname=None, country=None, language=None,
                 locked_until=None):
        self.uid = uid
        self.login = login
        self.password = password
        self.firstname = firstname
        self.lastname = lastname
        self.country = country
        self.language = language
        self.locked_until = locked_until

    def __eq__(self, other):
        is_type_eq = type(other) == type(self)
        is_uid_eq = self.uid == other.uid
        is_login_eq = self.login == other.login
        is_pass_eq = self.password == other.password
        if not is_type_eq:
            return False
        if is_uid_eq and is_login_eq and is_pass_eq:
            return True
        return False


def get_account_from_response(response):
    return Account(**response)


def get_exist_account_in_test():
    return Account(
        uid='4038895344',
        login='yandex-team-37493-63331',
        password='xPwD.sh5F'
    )


def get_exist_account_in_prod():
    return Account(
        uid='1048515902',
        login='yandex-team-60280-30120',
        password='T7gf.R3M5'
    )


def _random_numeric(n_digits):
    return ''.join(random.choice(string.digits) for _ in range(n_digits))


def generate_login():
    return 'yandex-team-{0}.{1}'.format(_random_numeric(5), _random_numeric(5))


def get_passport_environment_for_env(env):
    env = env.upper()
    if env == 'TEST':
        return 'testing'
    elif env == 'PROD':
        return 'production'
    else:
        raise KeyError('Unknown environment')
