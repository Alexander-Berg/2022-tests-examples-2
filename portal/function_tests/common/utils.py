# -*- coding: utf-8 -*-
import logging

import allure
from hamcrest import assert_that

logger = logging.getLogger(__name__)


def get_field(json_data, path, safe=False):
    if isinstance(path, basestring):
        path_list = path.split('.')
    elif isinstance(path, list):
        path_list = path
    else:
        if not safe:
            raise AttributeError('Invalid path for json: "{}"'.format(path))
        else:
            return None

    result = json_data
    current_path = []

    for p in path_list:
        current_path.append(p)
        if isinstance(result, dict) and p in result and result[p] is not None:
                result = result[p]
        elif isinstance(result, (list, tuple)):
            if p.isdigit() and int(p) < len(result):
                result = result[int(p)]
        else:
            logger.error('Failed to get path `{}` from json'.format('.'.join(current_path)))
            if not safe:
                raise KeyError('Failed to get path `{}` from json'.format('.'.join(current_path)))
            else:
                return None

    return result


def check_field(json_data, path, matcher, safe=False):
    with allure.step('Check field {}: {}'.format(path, matcher)):
        value = get_field(json_data, path, safe)

        assert_that(value, matcher, 'Invalid "{}" value'.format(path))

def delete(hashmap, key):
    value = None
    if isinstance(hashmap, dict) and key in hashmap:
        value = hashmap[key]
        del hashmap[key]
    return value

def dump_obj(obj):
    for attr in dir(obj):
        if hasattr( obj, attr ):
            print "obj.%s = %s" % (attr, getattr(obj, attr))

def parse_set_cookie(set_cookie_str = ""):
    # yandexuid=964148831652799476; Expires=Fri, 14-May-2032 14:57:55 GMT; Domain=.yandex.ru; Path=/; Secure; SameSite=None, yuidss=964148831652799476; Expires=Fri, 14-May-2032 14:57:55 GMT; Domain=.yandex.ru; Path=/; Secure; SameSite=None
    week = {
        "sun": 1,
        "mon": 1,
        "tue": 1,
        "wed": 1,
        "thu": 1,
        "fri": 1,
        "sat": 1,
    }
    set_cookie = []
    start = 0
    pos = set_cookie_str.find(",")
    while pos != -1:
        if pos - start > 3 and set_cookie_str[pos-4] == "=" and set_cookie_str[pos-3:pos].lower() in week:
            pos = set_cookie_str.find(",", pos + 1)
        else:
            set_cookie.append(set_cookie_str[start:pos])
            start = pos + 1
            if start >= len(set_cookie_str):
                break
            pos = set_cookie_str.find(",", start)
    if start < len(set_cookie_str):
        set_cookie.append(set_cookie_str[start:])

    return set_cookie

class MadmMockError(Exception):
    pass

def check_madm_error(data):
    if data is not None:
        try:
            assert 'madm_mock_error' not in data
        except AssertionError:
            raise MadmMockError('madm_mock_error={}'.format(data['madm_mock_error']))
