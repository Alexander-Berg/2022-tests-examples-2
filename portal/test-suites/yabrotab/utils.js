'use strict';

function setWidth (width, cols, ratio) {
    var page = document.querySelector('.b-page'),
        pageStyle = page.style,
        GAP = 8;

    const oneItem = (width - (cols - 1) * GAP) / cols,
        wide = cols > 4;

    page.classList.toggle('b-page_width_wide', wide);
    page.classList.toggle('b-page_width_normal', !wide);
    pageStyle.setProperty('--page-width', `${width}px`);
    pageStyle.setProperty('--item-ratio', ratio);
    pageStyle.setProperty('--item-width', `${oneItem}px`);
    pageStyle.setProperty('--cols-count', cols);
    if (window.broZen) {
        window.broZen.setWidth(+width);
    }
}

async function waitForPageReady() {
    await this.browser.yaInBrowser(async () => {
        await new Promise(resolve => {
            if (document.readyState === 'complete') {
                return resolve();
            }
            window.addEventListener('load', resolve);
        });

        await new Promise(resolve => {
            if (window.pageStatus.visible) {
                return resolve();
            }
            window.pageStatus.addEventListener('page-visible-changed', resolve);
        });
    });

    await this.browser.$("[class*='b-page_scrollable']").then(elem => elem.waitForExist({
        timeout: 5000
    }));
}

async function isBlocksRefreshedAfter(commands) {
    const TIMEOUT = 5000;
    const startTs = await this.browser.yaInBrowser(function getStartTs() {
        return document.querySelector('.main').dataset.ts;
    });

    try {
        await commands.call(this);
    } catch (e) {
        e.message = `isBlocksRefreshedAfter: ошибка при выполнении комманд: ${e.message}`;
    }
    await this.browser
        .yaInBrowser(function checkEndTs(startTs, TIMEOUT) {
            const ts = document.querySelector('.main').dataset.ts;

            if (ts === startTs) {
                return;
            }

            return new Promise((resolve, reject) => {
                const observer = new MutationObserver(list => list.map(onMutationItem));

                function onMutationItem(item) {
                    if (item.addedNodes && Array.prototype.some.call(item.addedNodes, node => node.closest('.main'))) {
                        const endTs = document.querySelector('.main').dataset.ts;

                        if (endTs !== startTs) {
                            observer.disconnect();
                            resolve();
                        }
                    }
                }

                setTimeout(() => {
                    observer.disconnect();
                    reject(new Error(`Блоки не обновлись за ${TIMEOUT}мс`));
                }, TIMEOUT);

                observer.observe(document.querySelector('.main').parentNode, {childList: true});
            });
        }, startTs, TIMEOUT);

}

async function isBlocksNotRefreshed() {
    const TIMEOUT = 5000;
    const {error} = await this.browser
        .executeAsync(function getTimestaps(TIMEOUT, done) {
            const observer = new MutationObserver(list => list.map(onMutationItem));

            function onMutationItem(item) {
                if (item.addedNodes && Array.prototype.some.call(item.addedNodes, node => node.closest('.main'))) {
                    observer.disconnect();
                    done({
                        error: 'Начался рефреш'
                    });
                }
            }
            setTimeout(() => {
                observer.disconnect();
                done({});
            }, TIMEOUT);

            observer.observe(document.querySelector('.main').parentNode, {childList: true});
        }, TIMEOUT);

    if (error) {
        throw new Error(error);
    }
}

module.exports = {
    setWidth,
    waitForPageReady,
    isBlocksRefreshedAfter,
    isBlocksNotRefreshed,
    ignore: [
        '.news',
        '.weather2',
        '.traffic',
        '.feed'
    ],
    hide: [
        '.dev-panel',
        '.panel-item',
        '.zen-app__logo',
        '.banner',
        // Новости могут быть двустрочными, растягивая контент
        '.list__item'
    ]
};
