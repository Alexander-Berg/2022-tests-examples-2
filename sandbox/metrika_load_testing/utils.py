import time

import infra.yp_service_discovery.python.resolver.resolver as yp_sd_resolver
import infra.yp_service_discovery.api.api_pb2 as yp_sd_api


def resolve_endpoints(endpoint_set_id, datacenter):
    resolver = yp_sd_resolver.Resolver(client_name='metrika-load-testing')

    request = yp_sd_api.TReqResolveEndpoints()
    request.cluster_name = datacenter
    request.endpoint_set_id = endpoint_set_id

    result = resolver.resolve_endpoints(request)
    return result.endpoint_set.endpoints


def get_tank_address(
    endpoint_set_id,
    datacenter,
    timeout=600,
    poll_period=15,
):

    host = None
    port = None

    start_ts = time.time()
    current_ts = time.time()

    while True:
        if (current_ts - start_ts) > timeout:
            raise ValueError('Failed to resolve tank endpoints: timeout reached ({}s)'.format(timeout))

        endpoints = resolve_endpoints(endpoint_set_id, datacenter)
        if endpoints:
            host = endpoints[0].fqdn
            port = endpoints[0].port
            break

        time.sleep(poll_period)
        current_ts = time.time()

    return host, port
