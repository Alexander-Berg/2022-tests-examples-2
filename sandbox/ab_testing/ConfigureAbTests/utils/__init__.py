import six  # noqa
import typing  # noqa
import logging
import os
import yaml
import json

from sandbox.projects.common.vcs import arc
from sandbox.projects.release_machine.core import const as rm_const


class ExperimentConfigLoadingError(Exception):
    pass


STR = six.string_types
STR_OR_INT = typing.Union[six.string_types + six.integer_types]

ARC_INFO_NAMES_KEY = "names"
ARC_INFO_NAMES_PATH_KEY = "path"


def get_config_path_from_arc_commit_info(commit_id, commit_info):  # type: (STR_OR_INT, STR) -> STR
    """
    Given a commit id and it's info returns experiment config path

    :raises ExperimentConfigLoadingError:
        Raised if unexpected commit info obtained (e.g., too much paths detected or no paths at all)
    """

    if len(commit_info) == 0:
        raise ExperimentConfigLoadingError(
            "Unable to get arc info for commit {}. `arc show` response was: {}".format(
                commit_id,
                commit_info,
            ),
        )

    commit_info = commit_info[0]

    if ARC_INFO_NAMES_KEY not in commit_info:
        raise ExperimentConfigLoadingError(
            "`arc show` result cannot be parsed (no {} key found): {}".format(ARC_INFO_NAMES_KEY, commit_info)
        )

    commit_info_names = commit_info[ARC_INFO_NAMES_KEY]

    if len(commit_info_names) > 1:
        raise ExperimentConfigLoadingError(
            "More than 1 path detected for commit {} - unable to decide which one to use. Full response: {}".format(
                commit_id,
                commit_info_names,
            ),
        )

    path = commit_info_names[0].get(ARC_INFO_NAMES_PATH_KEY, "")

    if not path:
        raise ExperimentConfigLoadingError("Path cannot be detected for commit {}".format(commit_id))

    return path


def load_experiment_config_from_fs(arcadia_root, config_path):  # type: (STR, STR) -> dict
    """
    Load experiment yaml config

    :param arcadia_root: Arcadia local root path
    :param config_path: config file path relative to Arcadia root
    :return: experiment config (dict)
    """

    from library.python import fs

    full_path = os.path.join(arcadia_root, config_path)
    logging.info("Full path to load: %s", full_path)
    return yaml.safe_load(fs.read_file_unicode(full_path))


def get_experiment_config(
    commit_id,
    arc_secret_name=rm_const.COMMON_TOKEN_NAME,
    arc_secret_owner=rm_const.COMMON_TOKEN_OWNER,
    arc_oauth_token=None,
):  # type: (STR_OR_INT, STR, STR, STR) -> dict
    """
    Load experiment config committed at :param commit_id:

    :param commit_id: either arc commit hash, or svn revision (both str and int accepted)
    :param arc_secret_name: arc secret name (SB Vault)
    :param arc_secret_owner: arc secret owner (SB Vault)
    :param arc_oauth_token: arc OAuth token

    :raises arc.ArcCommandFailed:
        Raised if arc command fails

    :raises ExperimentConfigLoadingError:
        Raised if unexpected commit info obtained (e.g., too much paths detected or no paths at all)

    """

    logging.info("Going to load experiment config")

    if isinstance(commit_id, six.integer_types) or commit_id.isdigit():
        commit_id = "r{}".format(commit_id)

    logging.info("commit: %s", commit_id)

    arc_client = arc.Arc(
        secret_name=arc_secret_name,
        secret_owner=arc_secret_owner,
        arc_oauth_token=arc_oauth_token,
    )

    with arc_client.mount_path(None, None, fetch_all=False, extra_params=["--vfs-version", "2"]) as mp:

        commit_info = arc_client.show(mp, commit=commit_id, as_dict=True, name_status=True)

        logging.info("`arc show` response: %s", commit_info)

        config_file_path = get_config_path_from_arc_commit_info(commit_id, commit_info)

        logging.info("Config file path detected: %s", config_file_path)

        arc_client.checkout(mp, commit_id)

        logging.info("Checked Arcadia out at %s", commit_id)

        config = load_experiment_config_from_fs(mp, config_file_path)

        logging.info("Config loaded: %s", json.dumps(config, indent=2))

    return config
