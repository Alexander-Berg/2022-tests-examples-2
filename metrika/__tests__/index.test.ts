import { hasOrderedInclusion } from '../utils';

describe('hasOrderedInclusion', () => {
    it('should return true', () => {
        expect(hasOrderedInclusion([['1', '2', '3'], ['1']])).toBeTruthy();
        expect(
            hasOrderedInclusion([
                ['1', '2', '3'],
                ['1', '2'],
            ]),
        ).toBeTruthy();
        expect(
            hasOrderedInclusion([['1', '2', '3'], ['2'], ['3'], ['2', '4']]),
        ).toBeTruthy();
    });

    it('should return false', () => {
        expect(
            hasOrderedInclusion([['1', '2', '3'], ['2'], ['3', '2']]),
        ).toBeFalsy();
        expect(
            hasOrderedInclusion([
                ['1', '2', '3'],
                ['1', '2', '4'],
            ]),
        ).toBeFalsy();
        expect(
            hasOrderedInclusion([
                ['1', '2', '3'],
                ['1', '3'],
                ['4', '2'],
            ]),
        ).toBeFalsy();
    });
});
