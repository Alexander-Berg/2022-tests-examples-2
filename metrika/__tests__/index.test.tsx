// import * as React from 'react';
// import { ReactElementsList } from 'typings/react';
// import { jestSnapshotRenderTest } from 'testing/jest-utils';
// import { CollapseToggler } from '..';

// const componentsForSnapshotRenderTest: ReactElementsList = {
//     'collapsed true': <CollapseToggler collapsed={true} />,
//     'collapsed false': <CollapseToggler collapsed={false} />,
//     'incoming className': <CollapseToggler className="className" />,
//     'size s': <CollapseToggler size="s" />,
// };

// describe('CollapseToggler', () => {
//     describe('renders', () => {
//         jestSnapshotRenderTest(componentsForSnapshotRenderTest);
//     });
// });
it('', () => {
    expect(true).toBe(true);
});
