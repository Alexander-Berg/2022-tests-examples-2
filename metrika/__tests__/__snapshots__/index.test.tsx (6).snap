// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help renders with hint prop 1`] = `
<Fragment>
  <Anchor>
    <styled.div>
      <styled.div>
        test
      </styled.div>
    </styled.div>
  </Anchor>
</Fragment>
`;
